Source/Events.Store.Services.Grpc/EventStoreGrpcService.cs

```diff
@@ -41,4 +43,51 @@ public override Task<CommitAggregateEventsResponse> CommitForAggregate(CommitAgg
     /// <inheritdoc/>
     public override Task<FetchForAggregateResponse> FetchForAggregate(FetchForAggregateRequest request, ServerCallContext context)
         => _eventStore.FetchAggregateEvents(request, context.CancellationToken);
+
+    public override async Task FetchForAggregateInBatches(FetchForAggregateInBatchesRequest request, IServerStreamWriter<FetchForAggregateResponse> responseStream, ServerCallContext context)
+    {
+        var eventSourceId = request.Aggregate.EventSourceId;
+        var aggregateRootId = request.Aggregate.AggregateRootId.ToGuid(); 
+        var fetchResult = request.RequestCase == FetchForAggregateInBatchesRequest.RequestOneofCase.FetchEvents
+            ? await _eventStore.FetchAggregateEvents(
+                eventSourceId,
+                aggregateRootId,
+                request.FetchEvents.EventTypes.Select(_ => _.ToArtifact()),
+                request.CallContext.ExecutionContext.TenantId.ToGuid(),
+                context.CancellationToken).ConfigureAwait(false)
+            : await _eventStore.FetchAggregateEvents(
+                eventSourceId,
+                aggregateRootId,
+                request.CallContext.ExecutionContext.TenantId.ToGuid(),
+                context.CancellationToken);
+        if (!fetchResult.Success)
+        {
+            var response = new FetchForAggregateResponse { Failure = fetchResult.Exception.ToFailure() };
+            await responseStream.WriteAsync(response).ConfigureAwait(false);
+            return;
+        }
+
+        await _aggregateEventsBatchSender.Send(
+            MaxBatchMessageSize,
+            fetchResult.Result.EventStream.GetAsyncEnumerator(context.CancellationToken),
+            () => CreateResponse(aggregateRootId, eventSourceId, fetchResult.Result.AggregateRootVersion),
+            (batch, aggregateEvent) => batch.Events.Events.Add(aggregateEvent.ToProtobuf()),
+            aggregateEvent => aggregateEvent.ToProtobuf(),
```

Comment:
See the comment on the `StreamOfBatchedMessagesSender` below - there are two calls to `aggregateEvent.ToProtobuf()` here - we really only need one.